drivers/pinctrl/pinctrl_ti_k3.c

```diff
@@ -6,31 +6,43 @@
 
 #define DT_DRV_COMPAT ti_k3_pinctrl
 
+#include <k3_mmr.h>
+#include <zephyr/device.h>
 #include <zephyr/arch/cpu.h>
 #include <zephyr/drivers/pinctrl.h>
 
 #define PINCTRL_NODE DT_NODELABEL(pinctrl)
+#define PADCFG_NODE  DT_NODELABEL(padcfg)
 
 static struct pinctrl_ti_k3_dev_data {
 	DEVICE_MMIO_RAM;
 } pinctrl_ti_k3_dev;
 
 static struct pinctrl_ti_k3_cfg_data {
 	DEVICE_MMIO_ROM;
-} pinctrl_ti_k3_cfg = {
-	DEVICE_MMIO_ROM_INIT(PINCTRL_NODE)
-};
+} pinctrl_ti_k3_cfg = {DEVICE_MMIO_ROM_INIT(PINCTRL_NODE)};
 
 int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintptr_t reg)
 {
 	ARG_UNUSED(reg);
+	const struct device *padcfg_dev = DEVICE_DT_GET_OR_NULL(PADCFG_NODE);
 	const struct device *dev = DEVICE_DT_GET(PINCTRL_NODE);
 	uintptr_t virt_reg_base = DEVICE_MMIO_GET(dev);
 
+	if (padcfg_dev) {
+		k3_unlock_mmr_partition(padcfg_dev, 0);
+		k3_unlock_mmr_partition(padcfg_dev, 1);
+	}
+
 	for (uint8_t i = 0; i < pin_cnt; i++) {
 		sys_write32(pins[i].value, virt_reg_base + pins[i].offset);
 	}
 
+	if (padcfg_dev) {
+		k3_lock_mmr_partition(padcfg_dev, 0);
+		k3_lock_mmr_partition(padcfg_dev, 1);
```

Comment:
Will this part == 0 or 1 be always true for all devices?

Reply:
It should be the case for all K3 devices, but I understand that a separate dt-binding might be in order for partition addresses